### PR TITLE
libx11 Update to 1.6.12, use clang linker

### PR DESCRIPTION
**Pull request by satmandu**
Part of https://github.com/skycocker/chromebrew/pull/4450

Works properly:
- [x] x86_64

### CODE REVIEW

File: packages/libx11.rb [hunk 1]
```diff
@@ -3,29 +3,20 @@
...
-  })
 
   depends_on 'xorg_proto'
   depends_on 'libxcb'
   depends_on 'libxtrans'
+  depends_on 'llvm' => ':build'
   
   def self.build
+    ENV['CFLAGS'] = "-fuse-ld=lld"
+    ENV['CXXFLAGS'] = "-fuse-ld=lld"
```
Comment (uberhacker) [inline packages/libx11.rb:19]: Does this break arm builds?

Comment (satmandu) [reply packages/libx11.rb:19]: Just got an armv7l-ish builder setup using this: https://gist.github.com/satmandu/791dd6e982d365dda6bbba5c332f0f73

Testing various compiles on it now.

File: packages/libx11.rb [hunk 2]
```diff
@@ -3,29 +3,20 @@
...
     system "./configure --prefix=#{CREW_PREFIX} --libdir=#{CREW_LIB_PREFIX}"
```
Comment (uberhacker) [inline packages/libx11.rb:20]: Use `#{CREW_OPTIONS}`?

File: packages/libx11.rb (hunk 1 above)
Comment (satmandu) [reply packages/libx11.rb:19]: works fine

File: packages/libx11.rb (hunk 2 above)
Comment (satmandu) [reply packages/libx11.rb]: done

**Pull request merged**